File: hardware/shapepoint.py
class ShapePoint:
    def __init__(self, lat = None, lon = None):
        self.lat = lat
        self.lon = lon

    def set_lat_if_less(self, lat):
        if (self.lat is None or lat < self.lat):
            self.lat = lat

    def set_lat_if_greater(self, lat):
        if (self.lat is None or lat > self.lat):
            self.lat = lat

    def set_long_if_less(self, lon):
        if (self.lon is None or lon < self.lon):
            self.lon = lon

    def set_long_if_greater(self, lon):
        if (self.lon is None or lon > self.lon):
            self.lon = lon

    def get_distance(self, p):
        return util.get_haversine_distance(self.lat, self.lon, p.lat, p.lon)

    def get_distance_from_lat_long(self, plat, plon):
        return util.get_haversine_distance(self.lat, self.lon, plat, plon)

    def __str__(self):
        return '(' + str(self.lat) + ', ' + str(self.lon) + ')'

File: server/app-engine/static/shapepoint.js
if(typeof util === 'undefined'){
    util = require('../static/gtfs-rt-util');
}
(function(exports) {
    exports.ShapePoint = class {
        constructor(lat = null, lon = null) {
            // util.log("ShapePoint constructor");
            if(lat !== null){
                this.lat = lat;
            } else this.lat = lat;
            if(lon !== null){
                this.lon = lon;
            } else this.lon = lon;
        }

        setLatIfLess(lat){
            let latFloat = lat;
            if (this.lat === null || latFloat < this.lat){
                this.lat = latFloat;
            }
        }

        setLatIfGreater(lat){
            let latFloat = lat;
            if (this.lat === null || latFloat > this.lat){
                this.lat = latFloat;
            }
        }

        setLonIfLess(lon){
            let lonFloat = lon;
            if (this.lon === null || lonFloat < this.lon){
                this.lon = lonFloat;
            }
        }

        setLonIfGreater(lon){
            let lonFloat = lon;
            if (this.lon === null || lonFloat > this.lon){
                this.lon = lonFloat;
            }
        }

        get_distance(p){
            return util.getHaversineDistance(this.lat, this.lon, p.lat, p.lon)
        }

        get_distance_from_lat_long(plat, plon){
            return util.getHaversineDistance(this.lat, this.lon, plat, plon)
        }

        getLat() {
            return this.lat;
        }
    }
}(typeof exports === 'undefined' ? this.shapepoint = {} : exports));
